src/AddIns/Uno.UI.MediaPlayer.WebAssembly/HtmlMediaPlayer.cs

```diff
@@ -320,27 +329,29 @@ private static void OnSourceChanged(DependencyObject dependencyobject, Dependenc
 	{
 		if (dependencyobject is HtmlMediaPlayer player)
 		{
+
+			player.TimeUpdated -= player.OnHtmlTimeUpdated;
+
 			var encodedSource = WebAssemblyRuntime.EscapeJs((string)args.NewValue);
 
 			if (player.Log().IsEnabled(LogLevel.Debug))
 			{
 				player.Log().Debug($"HtmlMediaPlayer.OnSourceChanged: {args.NewValue} isVideo:{player.IsVideo} isAudio:{player.IsAudio}");
 			}
 
-			ActiveElement = player.IsVideo ? _htmlVideo : player.IsAudio ? _htmlAudio : default;
-			ActiveElementName = player.IsVideo ? "Video" : player.IsAudio ? "Audio" : "";
+			player.ActiveElement = player.IsVideo ? player._htmlVideo : player.IsAudio ? player._htmlAudio : default;
```

Comment:
Let's be explicit about the precedence of statements here, same for the block below.
```suggestion
			player.ActiveElement = player.IsVideo 
						? player._htmlVideo 
						: (player.IsAudio 
									? player._htmlAudio 
									: default);
```